File: leetcode/solutions/0198.HouseRobber/rob_test.ts
import { test } from "https://deno.land/std/testing/mod.ts";
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import rob from "./rob.ts";

test("0198. House Robber", () => {
  assertEquals(rob([1,2,3,1]), 4);
  assertEquals(rob([2,7,9,3,1]), 12);
});

File: leetcode/solutions/0198.HouseRobber/rob_test.go
package solution

import (
	"testing"
)

func Test_0198_HouseRobber(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out int
	}{
		{
			[]int{1, 2, 3, 1},
			4,
		},
		{
			[]int{2, 7, 9, 3, 1},
			12,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := rob(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want  %#v", got, tt.out)
			}
		})
	}
}
